### PR TITLE
rec: Better detection of Bogus zone cuts for DNSSEC validation

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

`SyncRes::getDSRecords()` used to not set the `foundCut` flag as soon as the processing returned a `Bogus` state. In some situations, for example if the time was completely off and the `RRSIG` from the root zone were thus invalid, this wrong zone cut determination lead to validation issues.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
CircleCI failure is unrelated.

File: pdns/syncres.cc [hunk 1]
```diff
@@ -1747,58 +1747,63 @@ vState SyncRes::getDSRecords(const DNSName& zone, dsmap_t& ds, bool taOnly, unsi
...
 
+        /* delegation with no DS, might be Secure -> Insecure */
+        if (foundCut) {
           *foundCut = true;
         }
 
-        return Insecure;
-      } else if (foundCut && rcode == RCode::NoError && !ds.empty()) {
-        *foundCut = true;
+        return state == Secure ? Insecure : state;
```
Comment (omoerbeek) [inline pdns/syncres.cc]: This line can do with a comment

Comment (rgacogne) [reply pdns/syncres.cc]: I added some comments, let me know if that clarifies it for you!

Comment (omoerbeek) [reply pdns/syncres.cc]: Yes, nice.

**Pull request merged**